### PR TITLE
[stable/hackmd] bump hackmd (codimd)

### PR DESCRIPTION
**Pull request by kimxogus**
#### What this PR does / why we need it:
1. HackMD is renamed as CodiMD and env var should be renamed too.
2. session secrets and pod annotations are missing, so I added.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by kimxogus** [comment]
@paulczar rebased

File: stable/hackmd/templates/deployment.yaml [hunk 1]
```diff
@@ -47,8 +53,13 @@ spec:
               {{- else }}
               value: {{ .Values.postgresql.postgresPassword }}
               {{- end }}
-            - name: HMD_DB_URL
```
Comment (yuanying) [inline stable/hackmd/templates/deployment.yaml]: I guess `HMD_DB_URL` is still needed for codimd image v1.2.1. Are you testing this chart?

Comment (kimxogus) [reply stable/hackmd/templates/deployment.yaml]: Yes, it's deployed as company's private hackmd service and working fine

Comment (yuanying) [reply stable/hackmd/templates/deployment.yaml]: hmm, I checked script installed in 1.2.1 image.
It seems this image still depends on `HMD_DB_URL`...

```
docker run -ti --name test --entrypoint /bin/bash hackmdio/hackmd:1.2.1
root@821ef8d5b548:/hackmd# cat /usr/local/bin/docker-entrypoint.sh
#!/bin/sh

if [ "$HMD_DB_URL" = "" ]; then
    HMD_DB_URL="postgres://hackmd:hackmdpass@hackmdPostgres:5432/hackmd"
fi

export HMD_DB_URL

DB_SOCKET=$(echo ${HMD_DB_URL} | sed -e 's/.*:\/\//\/\//' -e 's/.*\/\/[^@]*@//' -e 's/\/.*$//')

if [ "$DB_SOCKET" != "" ]; then
    dockerize -wait tcp://${DB_SOCKET} -timeout 30s
fi

gosu hackmd node_modules/.bin/sequelize db:migrate

# Print warning if local data storage is used but no volume is mounted
[ "$HMD_IMAGE_UPLOAD_TYPE" = "filesystem" ] && { mountpoint -q ./public/uploads || {
    echo "
        #################################################################
        ###                                                           ###
        ###                        !!!WARNING!!!                      ###
        ###                                                           ###
        ###        Using local uploads without persistence is         ###
        ###            dangerous. You'll loose your data on           ###
        ###              container removal. Check out:                ###
        ###  https://docs.docker.com/engine/tutorials/dockervolumes/  ###
        ###                                                           ###
        ###                       !!!WARNING!!!                       ###
        ###                                                           ###
        #################################################################
    ";
} ; }

# Change owner and permission if filesystem backend is used
if [ "$HMD_IMAGE_UPLOAD_TYPE" = "filesystem" ]; then
    chown -R hackmd ./public/uploads
    chmod 700 ./public/uploads
fi

# Sleep to make sure everything is fine...
sleep 3

# run
exec gosu hackmd "$@"
```

Comment (SISheogorath) [reply stable/hackmd/templates/deployment.yaml]: Please make sure you checked the right repository. We switched to https://github.com/hackmdio/codimd-container

Comment (yuanying) [reply stable/hackmd/templates/deployment.yaml]: Yes I already checked the source in this repository but I think **the script which is installed in v1.2.1 image** still depends on HMD_DB_URL.

Comment (yuanying) [reply stable/hackmd/templates/deployment.yaml]: Also I checked **hackmdio/hackmd:1.2.1-alpine** image. This image depends on `HMD_DB_URL` too.

1.2.1-alpine image is released at `2018-10-11T10:03:55.213902564Z`, but `docker-entrypoint.sh` is modified to rename internals to CodiMD at `Fri Oct 12 11:58:08 2018 +0200` (see: https://github.com/hackmdio/codimd-container/commit/e3204e84bef9b9659efa1a7cd9ad579aa995b31c#diff-9a61e6f4e6c75734f99ceac894b85fac)

I think 1.2.1 image is slightly old.

Comment (SISheogorath) [reply stable/hackmd/templates/deployment.yaml]: You are correct, thanks for the heads up, I just pushed a fix, as soon as the build made it, it should be correct.

**Comment by yuanying** [comment]
I seems `hackmdio/hackmd:1.2.1` and `1.2.1-alpine` are rebased. So LGTM!!

File: stable/hackmd/templates/deployment.yaml [hunk 2]
```diff
@@ -38,7 +42,9 @@ spec:
               port: 3000
             initialDelaySeconds: 30
           env:
-            - name: HMD_DB_PASSWORD
+            - name: NODE_ENV
+              value: "production"
```
Comment (yuanying) [inline stable/hackmd/templates/deployment.yaml]: nit. NODE_ENV=production is already set by https://github.com/hackmdio/codimd-container/blob/master/debian/Dockerfile#L10 and https://github.com/hackmdio/codimd-container/blob/master/alpine/Dockerfile#L9

File: stable/hackmd/templates/deployment.yaml (hunk 1 above)
Comment (kimxogus) [reply stable/hackmd/templates/deployment.yaml]: Added `HMD_DB_URL`

File: stable/hackmd/templates/deployment.yaml (hunk 2 above)
Comment (kimxogus) [reply stable/hackmd/templates/deployment.yaml]: removed `NODE_ENV`

**Pull request merged**